### PR TITLE
:fire: Purge EnableMergeConflictDetection

### PR DESCRIPTION
**Pull request by say25**
## Overview

**Addresses #6184**

## Description

- Deletes EnableMergeConflictDetection Feature Flag with 🔥.

## Release notes
Notes: No Notes

### CODE REVIEW

File: app/src/ui/merge-branch/merge.tsx [hunk 1]
```diff
@@ -320,9 +283,7 @@ export class Merge extends React.Component<IMergeProps, IMergeState> {
           />
         </DialogContent>
         <DialogFooter>
-          {enableMergeConflictDetection()
-            ? this.renderNewMergeInfo()
-            : this.renderOldMergeMessage()}
+          {this.renderNewMergeInfo()}
```
Comment (say25) [inline app/src/ui/merge-branch/merge.tsx]: Should this be renamed to `renderNewMergeInfo`?

Comment (shiftkey) [reply app/src/ui/merge-branch/merge.tsx]: @say25 yes please!

**Review by shiftkey — DISMISSED** [review]

**Comment by shiftkey** [comment]
@say25 looks like there are some merge conflicts now with #6251 being merged

**Comment by say25** [comment]
@shiftkey conflict resolved

**Review by shiftkey — APPROVED** [review]

**Pull request merged**